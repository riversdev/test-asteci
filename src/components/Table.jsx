import { useState } from 'react'
import { useFetch } from '../hooks/useFetch'
import { useSingleFetch } from '../hooks/useSingleFetch'
import { Condition } from './Condition'

export const Table = () => {
    const { conditions, isLoading, hasError } = useFetch()
    const [current, setCurrent] = useState()
    const currentCondition = useSingleFetch(current)

    const { pagination, results } = conditions

    if (isLoading === true) {
        return (
            <p>Cargando...</p>
        )
    }

    if (hasError !== null) {
        return (
            <p>Ooops. Error de la api !</p>
        )
    }

    return (
        <>
            <Condition {...currentCondition} />
            <h3>Total de registros: {results.length}</h3>
            <table>
                <thead>
                    <tr>
                        <th>_id</th>
                        <th>cityid</th>
                        <th>name</th>
                        <th>state</th>
                        <th>probabilityofprecip</th>
                        <th>relativehumidity</th>
                        <th>Lastreporttime</th>
                        <th>LLUEVE</th>
                    </tr>
                </thead>
                <tbody>
                    {results.map(({ _id, cityid, name, state, probabilityofprecip, relativehumidity, lastreporttime }) => (
                        <tr
                            key={_id}
                        >
                            <td onClick={() => setCurrent(_id)}>{_id}</td>
                            <td>{cityid}</td>
                            <td>{name}</td>
                            <td>{state}</td>
                            <td>{probabilityofprecip}</td>
                            <td>{relativehumidity}</td>
                            <td>{lastreporttime}</td>
                            <td>{probabilityofprecip > 60 || relativehumidity > 50 ? 'SI' : 'NO'}</td>
                        </tr>
                    ))}
                </tbody>
            </table>
        </>
    )
}