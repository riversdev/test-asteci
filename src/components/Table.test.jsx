import { render, screen } from '@testing-library/react'
import { Table } from './Table'

describe('tests Table', () => {
    test('should display the table', () => {
        const { container } = render(<Table />)

        expect(screen.getAllByText('Cargando...')).toBeTruthy()

        expect(container).toMatchSnapshot()

        // screen.debug()
    })
})