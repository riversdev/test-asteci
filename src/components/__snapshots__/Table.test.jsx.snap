// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Table should display the table 1`] = `
<div>
  <p>
    Cargando...
  </p>
</div>
`;
